elements/lisk-tree/src/merkle_tree.ts

```diff
@@ -154,200 +150,65 @@ export class MerkleTree {
 		return this.root;
 	}
 
-	public generatePath(_queryData: ReadonlyArray<Buffer>): Path {
+	public generateProof(queryData: ReadonlyArray<Buffer>): Proof | undefined {
 		if (this._width === 1) {
-			return [];
+			return undefined;
 		}
-		const treeStructure = this._getStructure();
+		const treeStructure = this._getPopulatedStructure();
 		const path = [];
-		// Get full node info of all query nodes
-		const queryNodes = [];
-		for (let i = 0; i < _queryData.length; i += 1) {
-			try {
-				const queryNode = this.getNode(_queryData[i]);
-				queryNodes.push(queryNode);
-			} catch (err) {
-				queryNodes.push(undefined);
+		const addedPath = new Set();
+		const indexes = [];
+		let queryNode: NodeInfo | undefined;
+
+		for (let i = 0; i < queryData.length; i += 1) {
+			queryNode = this.getNode(queryData[i]);
```

Comment:
what happen if the non existent queryData is provided?

Reply:
It will throw an error which can be caught in the outer scope.

Reply:
In the LIP we assign a special flag index (0) to any queryData that is not part of the tree, so that the verifier can discard them